travis/test_pylint

```diff
@@ -44,163 +44,467 @@ Use the following types of pylint configuration files:
-    if os.path.isfile(version_cfg):
+CLICK_DIR = click.Path(exists=True, dir_okay=True, resolve_path=True)
+EXPECTED_ERRORS = {
+    'openerp-exception-warning': 4,
+    'relative-import': 1,
+    'redundant-modulename-xml': 1,
+    'attribute-deprecated': 4,
+    'no-method-argument': 5,
+    'api-one-multi-together': 1,
+    'print-statement': 1,
+    'translation-field': 2,
+    'sql-injection': 1,
+    'reimported': 4,
+    'wrong-tabs-instead-of-spaces': 1,
+    'xml-syntax-error': 1,
+    'assignment-from-none': 1,
+    'anomalous-backslash-in-string': 1,
+    'wrong-import-order': 4,
+    'manifest-required-key': 1,
+    'pointless-string-statement': 1,
+    'unreachable': 1,
+    'too-complex': 1,
+    'missing-manifest-dependency': 2,
+    'missing-newline-extrafiles': 3,
+    'incoherent-interpreter-exec-perm': 2,
+    'no-utf8-coding-comment': 1,
+    'duplicate-xml-record-id': 1,
+    'missing-readme': 2,
+    'missing-import-error': 2,
+    'rst-syntax-error': 1,
+    'use-vim-comment': 1,
+    'dangerous-default-value': 1,
+    'translation-required': 1,
+    'api-one-deprecated': 3,
+    'invalid-commit': 1,
+    'docstring-first-line-empty': 1,
+    'class-camelcase': 1,
+    'duplicate-xml-fields': 1,
+    'manifest-deprecated-key': 2,
+    'file-not-used': 2,
+    'method-required-super': 5,
+    'duplicate-id-csv': 1,
+    'deprecated-module': 3,
+    'redundant-keyword-arg': 1,
+    'redefined-builtin': 2,
+    'pointless-statement': 1,
+    'return-in-init': 1,
+    'copy-wo-api-one': 2,
+    'manifest-version-format': 1,
+    'duplicate-key': 2,
+    'manifest-required-author': 5,
+    'too-few-format-args': 1}
+
+
+@click.command()
+@click.option(
+    'paths', '--path', envvar='TRAVIS_BUILD_DIR', multiple=True,
+    type=CLICK_DIR, required=True, help="Addons paths to check pylint")
+@click.option(
+    '--config-file', '-c', type=click.File('r', lazy=True), required=True,
+    help="Pylint config file")
+@click.option(
+    '--sys-paths', '-sys-path', envvar='PYTHONPATH', multiple=True,
+    type=CLICK_DIR, help="Additional paths to append in sys path.")
+@click.option(
+    '--extra-params', '-extra-param', multiple=True,
+    help="Extra pylint params to append in pylint command")
+@click.option(
+    '--msgs-no-count', '-msgs-no-count', multiple=True,
+    help="List of messages that will not add to the failure count.")
+def get_cmd(paths, config_file, msgs_no_count=None,
+            sys_paths=None, extra_params=None):
+    if sys_paths is None:
+        sys_paths = []
+    if extra_params is None:
+        extra_params = []
+    sys.path.extend(sys_paths)
+    cmd = ['--rcfile=' + config_file.name]
+    cmd.extend(extra_params)
+    subpaths = run_pylint.get_subpaths(list(paths))
+    if not subpaths:
+        message = "Python modules not found in paths %s" % (paths)
+        raise UserWarning(message)
+    cmd.extend(subpaths)
+    return cmd, list(msgs_no_count)
+
+
+class MainTest(unittest.TestCase):
```

Comment:
This script is the core of pylint tests from other projects.
I don't understand why is need migrate it for `unittest.TestCase`
The script `self_test` is a good idea migrate it for `unitttest.TestCase` and manage all test cases there.
But the main script shouldn't have changes because we want improvement the tests don't the self-script.